src/Symfony/Component/DependencyInjection/Compiler/AttributeAutoconfigurationPass.php

```diff
@@ -0,0 +1,53 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\DependencyInjection\Compiler;
+
+use Symfony\Component\DependencyInjection\ContainerBuilder;
+
+/**
+ * @author Alexander M. Turek <me@derrabus.de>
+ */
+final class AttributeAutoconfigurationPass implements CompilerPassInterface
+{
+    public function process(ContainerBuilder $container): void
+    {
+        if (80000 > \PHP_VERSION_ID) {
+            return;
+        }
+
+        $autoconfiguredAttributes = $container->getAutoconfiguredAttributes();
+
+        foreach ($container->getDefinitions() as $definition) {
+            if (!$definition->isAutoconfigured()) {
+                continue;
+            }
+
+            if (!$class = $container->getParameterBag()->resolveValue($definition->getClass())) {
+                continue;
+            }
+
+            try {
+                $reflector = new \ReflectionClass($class);
+            } catch (\ReflectionException $e) {
+                continue;
+            }
+
+            foreach ($reflector->getAttributes() as $attribute) {
```

Comment:
we're also going to have to ensure that resource tracking works, ie that the container is refreshed when an attribute is updated

Reply:
> we would look for arguments on the constructor, on setters, and on public properties

For now, I did descend into the structural element of each class because I feared the performance impact might be too heavy.

A possible way we could solve this is by using a class-level attribute and setting use `registerAttributeForAutoconfiguration()` to register a closure that looks for more attributes on methods/properties/constructor parameters.

Reply:
Maybe we don't have to solve this now. We just need to be sure we won't force us into BC breaks when we'll want to support attributes on eg arguments.